ygen/codegen.go

```diff
@@ -885,11 +896,25 @@ type mappedYANGDefinitions struct {
 // It returns a mappedYANGDefinitions struct populated with the directory, enum
 // entries in the input schemas as well as the calculated schema tree.
 func mappedDefinitions(yangFiles, includePaths []string, cfg *GeneratorConfig) (*mappedYANGDefinitions, util.Errors) {
-	modules, errs := processModules(yangFiles, includePaths, cfg.ParseOptions.YANGParseOptions)
+	moduleEntries, modules, errs := processModules(yangFiles, includePaths, cfg.ParseOptions.YANGParseOptions)
 	if errs != nil {
 		return nil, errs
 	}
 
+	// Exclude any modules that do not appear in the yangFiles list.
+	if cfg.ExcludeSearchPathModules {
+		inFiles := map[string]bool{}
+		for _, f := range yangFiles {
+			inFiles[f] = true
+		}
+		for _, v := range modules {
+			moduleSrc := strings.Split(v.Source.Location(), ":")[0]
```

Comment:
nit: rename to `moduleSrcFile`. This is a good hack, since there is no other way to find the name of the file given a module. Could you also just add a TODO for me?

In case `moduleSrc` is the empty string, I think we should bail.